src/main/java/org/example/Phone.java

```
@@ -0,0 +1,112 @@
+     * You can display the call history, retrieve first or second message for each contact.
+     * Each call or message drains the battery and you can use getBatteryLife() to check current levels.
+     */
+    protected final int batteryLife;
+    protected String color;
+    protected String material;
+    protected String imei;
+    protected int currentBatteryLife;
+    LinkedHashMap<String, Contact> contactList = new LinkedHashMap<>();
+    private ArrayList<String> callHistory = new ArrayList<>();
+
+    protected Phone(int batteryLife, String color, String material, String imei) {
+        this.batteryLife = batteryLife;
+        this.color = color;
+        this.material = material;
+        this.imei = imei;
+        currentBatteryLife = batteryLife;
+    }
+
+    public void addContact(int id, String phoneNumber, String firstName, String lastName) {
+        Contact contact = new Contact(id, phoneNumber, firstName, lastName);
+        contactList.put(phoneNumber, contact);
+    }
+
+    public void printContacts() {
+        for (Map.Entry<String, Contact> set : contactList.entrySet()) {
+            System.out.println(set.getValue());
+        }
+    }
+
+    public String sendMessage(String phoneNumber, String content) {
+        if (contactList.containsKey(phoneNumber)) {
+            Message newMessage = new Message(content);
+            contactList.get(phoneNumber).addMessage(newMessage);
+            currentBatteryLife -= 1;
+            return "Message sent";
+        } else {
+            System.out.println("This number does not exist in the Contact List");
+            return "Message not sent";
+        }
+    }
+
+    public String getFirstMessage(String phoneNumber) {
+        if (contactList.containsKey(phoneNumber)) {
+            return contactList.get(phoneNumber).messages.get(0).getContent();
+        } else {
+            System.out.println("Contact not in list");
```

same as above, use the fail fast aproach, if contact not it list, exit directly